testnet/stacks-node/src/nakamoto_node/stackerdb_listener.rs

```diff
@@ -597,4 +699,31 @@ impl StackerDBListenerComms {
         // tenure.
         u64::MAX
     }
+
+    /// Get the transactions that at least 70% of the signing power expect to be replayed in
+    /// the next stacks block
+    pub fn get_replay_transactions(&self, weight_threshold: u32) -> Vec<StacksTransaction> {
```

Comment:
I think I'd rather see this get abstracted away from the signer update messages somewhat. Instead of iterating over the signer's updates and then weighting them, I think we should have something that looks kind of like:

```rust
let global_state = get_signer_global_state()?;
return Ok(global_state.replay_transactions)
```

So that the replay transactions are really just a field of the globally agreed upon state. I think this is nearly equivalent to what you have here, but I think it'll end up making our lives simpler if the global state machine just provides the replay transactions.